src/parser/paladin/protection/modules/talents/Seraphim.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent } from 'parser/core/Events';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import { formatNumber, formatPercentage } from 'common/format';
+import UptimeIcon from 'interface/icons/Uptime';
+import CriticalStrikeIcon from 'interface/icons/CriticalStrike';
+import VersatilityIcon from 'interface/icons/Versatility';
+import HasteIcon from 'interface/icons/Haste';
+import MasteryIcon from 'interface/icons/Mastery';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+
+const STAT_MODIFIER = 0.08;
+
+type BonusStats = {haste: number, crit: number, versatility: number, mastery: number};
+
+/**
+ * Consumes 3 Holy Power to grant 8% Haste, Crit, Versatility, and Mastery for 15 seconds.
+ */
+class Seraphim extends Analyzer {
+  static dependencies = {
+    statTracker: StatTracker,
+  };
+
+  protected statTracker!: StatTracker;
+
+  bonusStatGains: BonusStats[] = [];
+
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.SERAPHIM_TALENT.id);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.SERAPHIM_TALENT), this.onCast);
+    const localStatTracker = (options.statTracker as StatTracker);
+    const bonusStats = this.getBonusStats(localStatTracker);
+    localStatTracker.add(SPELLS.SERAPHIM_TALENT.id, bonusStats);
+
+  }
+
+  get uptime(): number {
+    return this.selectedCombatant.getBuffUptime(SPELLS.SERAPHIM_TALENT.id) / this.owner.fightDuration;
+  }
+
+  get averageMasteryGain(): number {
+    return this.bonusStatGains.map((gain) => gain.mastery).reduce((sum, next) => sum + next, 0) / this.bonusStatGains.length;
+  }
+
+  get averageCritGain(): number {
+    return this.bonusStatGains.map((gain) => gain.crit).reduce((sum, next) => sum + next, 0) / this.bonusStatGains.length;
+  }
+
+  get averageVersatilityGain(): number {
+    return this.bonusStatGains.map((gain) => gain.versatility).reduce((sum, next) => sum + next, 0) / this.bonusStatGains.length;
+  }
+
+  get averageHasteGain(): number {
+    return this.bonusStatGains.map((gain) => gain.haste).reduce((sum, next) => sum + next, 0) / this.bonusStatGains.length;
+  }
+
+  onCast(event: CastEvent): void {
+    this.bonusStatGains.push(this.getBonusStats(this.statTracker));
```

Comment:
what about stat DRs?

(i don't know what code we have in place to handle those at this time---boy does it complicate things tho)

Reply:
The stat DRs only apply to the amount of rating required per percentage increase, yeah? Currently this records the raw rating gain rather than percentage gain, so we shouldn't need to compensate for stat DRs.

Reply:
My thinking had been that we don't want to show raw rating gains without context for how effective those ratings actually were, but per my comment above i think that is a moot point for Sera.

Reply:
Updated Seraphim with displaying the percentage gain rather than raw rating gain:

<img width="782" alt="2020-10-29 11_20_29-Mythic Wrathion - Kill (4_22) by Decix in Ny'alotha - WoWAnalyzer" src="https://user-images.githubusercontent.com/6363390/97616354-ab737400-19d9-11eb-9ee0-6cfd0c0fb15a.png">